Adds wooden chapel pews

Caiggas opened the pull request:
<img width="325" alt="pewscreenshotfinal" src="https://user-images.githubusercontent.com/2732029/52499554-b0f7ad80-2ba9-11e9-8c19-35766cd29ec8.PNG">
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

This adds wooden chapel pews to the game. They function much like ratty couches, in that there are several different pieces which can be mixed and matched to make different length pews. They can be built ingame from 3 wooden sheets or placed by a mapper. They can also be deconstructed with a wrench into 3 wooden sheets. Pews are flammable.

This is my first PR, so please let me know if I messed up somewhere. Also, I did test and confirmed that it does compile.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

Just adds some chapel themed chairs, basically. Its entirely an RP thing.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: Wooden pews may be constructed from 3 wood for all your religious needs!
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Caiggas committed:
Adds wooden chapel pews

This adds wooden chapel pews to the game. They function much like ratty couches, in that there are several different pieces which can be mixed and matched to make different length pews. They can be built from 3 wooden sheets.

The change to sheet_types.dm adds the pew crafting recipies to the end of the wood sheets crafting list.

The added file pew.dm  is the code which makes the pews work.

The added file pew.dmi is the sprites for the pews, including the overlays for the armrests.

the change to tgstation.dme is a single line which includes pew.dm.

This is my first PR ever, so please let me know if I made any mistakes in approved process.

Tlaltecuhtli commented:
why is own .dmi

Caiggas commented:
I suppose I could merge it into something else. maybe chairs? I was following patterns established by similar items. Ratty couches have their own dmi, so I figured it was the correct way to do it. I apologize if I was incorrect. Let me know what dmi I should put this into, and Ill make the changes. Thanks for the feedback.

ExcessiveUseOfCobblestone commented:
the same as couches should be fine, you can rename the file if it bothers you (just update code)

Caiggas commented:
So, I apologize for my ignorance, but do I just close this PR, make my changes, and then open a new one? or is there a way for me to "overwrite" this PR with a corrected branch?

ExcessiveUseOfCobblestone commented:
you just make the changes on your `Add-Chapel-Pews` branch and they'll get transferred to here.

timothymtorres commented:
Those are some pretty nice sprites.  Well done!

Caiggas committed:
merged pew.dmi into sofa.dmi

As per the instructions of several commenters, I merged pew.dmi into sofa.dmi. pew.dm has been updated to reflect this.

Caiggas commented:
Alright, so I merged pew.dmi into sofa.dmi and updated pew.dm. I then mapped a pew in and tested it to confirm everything still works.

Caiggas commented:
so, travis ci fails on the "build all maps" step. I don't really understand what this is... Can anyone help me fix this? Or is this not something I need to worry about?

optimumtact commented on code/game/objects/items/stacks/sheets/sheet_types.dm:
```diff
@@ -205,6 +205,13 @@ GLOBAL_LIST_INIT(wood_recipes, list ( \
 	new/datum/stack_recipe("wooden crate", /obj/structure/closet/crate/wooden, 6, time = 50, one_per_turf = TRUE, on_floor = TRUE),\
 	new/datum/stack_recipe("baseball bat", /obj/item/melee/baseball_bat, 5, time = 15),\
 	new/datum/stack_recipe("loom", /obj/structure/loom, 10, time = 15, one_per_turf = TRUE, on_floor = TRUE), \
+	null, \
```
what's with the null entries?

Caiggas replied:
They add lines in the build menu to group items in the build menu. I copied the syntax from the sofa build options higher up in the file. I can remove them if you want.

optimumtact replied:
no that's fine, didn't realise they were used for formatting

optimumtact commented on code/game/objects/structures/beds_chairs/pew.dm:
```diff
@@ -0,0 +1,72 @@
+/obj/structure/chair/pew/left/Initialize()
+	leftpewarmrest = GetLeftPewArmrest()
+	leftpewarmrest.layer = ABOVE_MOB_LAYER
+	return ..()
```
you can just do `. = ..()`

`.` is the default return value of a proc

Caiggas replied:
gotcha, thanks for the info. I copied that syntax from comfy chairs.

Caiggas replied:
to be honest, I feel like there has to be a way to write that code in the pew section, and have it work in pew/left and pew/right, but I don't fully understand byond syntax all that well. I feel like having essentially identical code in pew/left and pew/right is dirty, but I wan't sure how to do it better.

ExcessiveUseOfCobblestone replied:
```DM
//under the base type
var/armrest_mut_appearance_key
var/mutable_appearance/the_armrest

//under the subtypes
armrest_mut_appearance_key = "pewend_left_armrest" //or "pewend_right_armrest" 

//Procs
/obj/structure/chair/pew/Initialize()
    . = ..()
    if(armrest_mut_appearance_key)
        the_armrest = mutable_appearance('icons/obj/sofa.dmi', armrest_mut_appearance_key)
        the_armrest.layer = ABOVE_MOB_LAYER

/obj/structure/chair/pew/Destroy()
    if(the_armrest)
        QDEL_NULL(the_armrest)
    return ..()

/obj/structure/chair/pew/post_buckle_mob(mob/living/M)
	. = ..()
	update_armrest()

/obj/structure/chair/pew/post_unbuckle_mob()
	. = ..()
	update_armrest()

/obj/structure/chair/pew/proc/update_armrest()
    if(the_armrest)
        if(has_buckled_mobs())
		    add_overlay(the_armrest)
	    else
		    cut_overlay(the_armrest)
```

PKPenguin321 commented:
Nice. Always found it weird that the chapel had plain metal chairs all over the place

Caiggas commented:
same here, I was mapping for fun, and I realized that the chapel didn't feel like a religious space.

Jimmius commented:
can't wait for the upcoming chapel rework where it has it's own minigames and systems that interact with the rest of the station, rather than just being a cool place to cremate bodies

JJRcop commented:
In this case travis failed on cdd4cc165dc6e6d038ec4e0f3ee555ec6488e52e because of an internal error unrelated to your code, however in the future, you can click on each build job and get a log from the worker, so you can see the compile errors that had caused it to fail. This is a very valuable resource to remember. You can check appveyor too. If it fails on travis it probably won't pass on appveyor either.

Cyprex commented:
The part between the CL is a summary of your changes for players, as such, consider something like "You can now build wooden benches for all your religious needs with 3 wood".

optimumtact commented:
The changelog should be a summary for the players and try to avoid putting any weird formatting in it. It doesn't handle much outside of plain text.

Caiggas commented:
Fixed it, thanks

JJRcop commented:
The add labels must be in the cl tags, you shouldn't put anything else in there besides tags. Here's a legend of the available tags: https://github.com/tgstation/tgstation/blob/master/.github/PULL_REQUEST_TEMPLATE.md#changelog

ExcessiveUseOfCobblestone commented:
someone edit this to have my code it's as close to free GBP as you're gonna get lol

Caiggas commented:
Yeah, I'm gonna update it when I get the motivation. What is GBP? My brain is drawing a blank.

ExcessiveUseOfCobblestone commented:
it's coder Good Boy Points
